### ISSUE TITLE
Pod::Man vs vendor_perl (wrong manpage titles)

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#7846]
Migrated from [rt.perl.org#34500](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=34500) (status was 'resolved')

Searchable as RT34500$

**Comment by p5pRT** [comment]
### From at@altlinux.ru

#### Created by at@altlinux.ru



Hello\,

When I run "perldoc LWP"\, I get wrong manpage title&#8203;: "vendor\_perl&#8203;::LWP"\.
\(The same way I get "vendor\_perl&#8203;::XML&#8203;::Parser" for XML&#8203;::Parser etc\.\)
This is because these modules are installed in vendor directories \(widely
used in modern Linux distributions for vendor\-supplied perl packages\)\.
Vendor directories were introduced with perl\-5\.8\.0\, and some pieces
of software are not ready for this change\.

Here is a patch for Pod/Man\.pm that fixes the problem for me&#8203;:

<details><summary>Inline Patch</summary>

```diff
--- perl-5.8.6/lib/Pod/Man.pm-	2003-03-31 11:16:36 +0000
+++ perl-5.8.6/lib/Pod/Man.pm	2005-03-19 18:58:12 +0000
@@ -411,7 +411,7 @@ sub begin_pod {
             }
             if ($cut > 0) {
                 splice (@dirs, 0, $cut);
-                shift @dirs if ($dirs[0] =~ /^site(_perl)?$/);
+                shift @dirs if ($dirs[0] =~ /^(site|vendor)(_perl)?$/);
                 shift @dirs if ($dirs[0] =~ /^[\d.]+$/);
                 shift @dirs if ($dirs[0] =~ /^(.*-$^O|$^O-.*|$^O)$/);
             }

```
</details>
End of patch

Now I get correct manpage titles\.
I also noticed similar issue in Pod/Find\.pm&#8203;:

<details><summary>Inline Patch</summary>

```diff
--- perl-5.8.6/lib/Pod/Find.pm-	2004-06-01 12:41:24 +0000
+++ perl-5.8.6/lib/Pod/Find.pm	2005-03-19 19:10:56 +0000
@@ -168,10 +168,10 @@ sub pod_find
 
         if ($^O eq 'MacOS') {
             $SIMPLIFY_RX =
-              qq!^(?i:\:?site_perl\:|\:?pod\:(?=.*?\\.pod\\z))*!;
+              qq!^(?i:\:?(?:site|vendor)_perl\:|\:?pod\:(?=.*?\\.pod\\z))*!;
         } else {
             $SIMPLIFY_RX =
-              qq!^(?i:site(_perl)?/|\Q$Config::Config{archname}\E/|\\d+\\.\\d+([_.]?\\d+)?/|pod/(?=.*?\\.pod\\z))*!;
+              qq!^(?i:(?:site|vendor)(_perl)?/|\Q$Config::Config{archname}\E/|\\d+\\.\\d+([_.]?\\d+)?/|pod/(?=.*?\\.pod\\z))*!;
         }
     }

```
</details>
End of patch 

This two files are known to be part of "podlators" and "PodParser" CPAN
distributions\.  Should I file bugs at rt\.cpan\.org too?


\-\- 
Alexey Tourbin
ALT Linux Team

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=low

Site configuration information for perl v5.8.6:

Configured by ALT Linux Team at Fri Mar 11 15:59:16 MSK 2005.

Summary of my perl5 (revision 5 version 8 subversion 6) configuration:
  Platform:
    osname=linux, osvers=2.4.20-alt7-up, archname=i386-linux-thread-multi
    uname='linux solemn.turbinal.org 2.4.20-alt7-up #1 fri mar 14 14:57:05 msk 2003 i686 unknown unknown gnulinux '
    config_args='-de -rs -Darchname=i386-linux -Dd_dosuid -Ud_csh -Dlibswanted=dl m c crypt db ndbm gdbm -Duseshrplib -Dlibperl=libperl.so.5.8 -Dcc=gcc -Doptimize=-pipe -Wall -Os -march=i686 -D_GNU_SOURCE -momit-leaf-frame-pointer -Dcccdlflags=-fPIC -DPIC -Dccdlflags=-rdynamic -Wl,-O1 -Dlddlflags=-shared -Wl,-O1 -Dldflags=-Wl,-O1 -Dprefix=/usr -Dprivlib=/usr/lib/perl5 -Darchlib=/usr/lib/perl5/i386-linux -Dvendorprefix=/usr -Dvendorlib=/usr/lib/perl5/vendor_perl -Dvendorarch=/usr/lib/perl5/vendor_perl/i386-linux -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dvendorman1dir=/usr/share/man/man1 -Dvendorman3dir=/usr/share/man/man3 -Dsiteprefix=/usr/local -Dsitelib=/usr/local/lib/perl5/site_perl/5.8.6 -Dsitearch=/usr/local/lib/perl5/site_perl/5.8.6/i386-linux -Dsiteman1dir=/usr/local/man/man1 -Dsite_man3dir=/usr/local/man/man3 -Dcf_by=ALT Linux Team -Dcf_email=qa@altlinux.org -Dmyhostname=localhost -Dperladmin=root@localhost -Dmyuname=Linux 2.4.20-alt7-up i686 -Dnewmyuname=Linux 2.4.20-alt7-up i686 -Dinc_version_list=5.8.5/i386-linux 5.8.4/i386-linux 5.8.3/i386-linux 5.8.2/i386-linux 5.8.1/i386-linux 5.8.0/i386-linux 5.8.5 5.8.4 5.8.3 5.8.2 5.8.1 5.8.0 5.6.1 5.6.0 -Dpager=/usr/bin/less -isR -Di_shadow -Di_syslog -Dusethreads -Duseithreads -Duselargefiles -Di_db -Di_gdbm -Di_ndbm -Di_sdbm -Ui_odbm'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DTHREADS_HAVE_PIDS -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm',
    optimize='-pipe -Wall -Os -march=i686 -D_GNU_SOURCE -momit-leaf-frame-pointer',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DTHREADS_HAVE_PIDS -fno-strict-aliasing -pipe -I/usr/local/include -I/usr/include/gdbm'
    ccversion='', gccversion='3.3.3 20040412 (ALT Linux, build 3.3.3-alt5)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-Wl,-O1 -L/usr/local/lib64 -L/usr/local/lib'
    libpth=/usr/local/lib64 /usr/local/lib /lib64 /usr/lib64 /lib /usr/lib
    libs=-ldl -lm -lpthread -lc -lcrypt -ldb -lgdbm
    perllibs=-ldl -lm -lpthread -lc -lcrypt
    libc=/lib/libc-2.3.3.so, so=so, useshrplib=true, libperl=libperl.so.5.8
    gnulibc_version='2.3.3'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic -Wl,-O1'
    cccdlflags='-fPIC -DPIC', lddlflags='-shared -Wl,-O1 -L/usr/local/lib64 -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.8.6:
    /etc/perl5
    /usr/lib/perl5/i386-linux
    /usr/lib/perl5
    /usr/local/lib/perl5/site_perl/5.8.6/i386-linux
    /usr/local/lib/perl5/site_perl/5.8.6
    /usr/local/lib/perl5/site_perl/5.8.4/i386-linux
    /usr/local/lib/perl5/site_perl/5.8.2/i386-linux
    /usr/local/lib/perl5/site_perl/5.8.1/i386-linux
    /usr/local/lib/perl5/site_perl/5.8.4
    /usr/local/lib/perl5/site_perl/5.8.2
    /usr/local/lib/perl5/site_perl/5.8.1
    /usr/local/lib/perl5/site_perl/5.6.1
    /usr/local/lib/perl5/site_perl
    /usr/lib/perl5/vendor_perl/i386-linux
    /usr/lib/perl5/vendor_perl
    /usr/lib/perl5/vendor_perl
    .


Environment for perl v5.8.6:
    HOME=/home/at
    LANG=C
    LANGUAGE (unset)
    LC_COLLATE=ru_RU.CP1251
    LC_CTYPE=ru_RU.CP1251
    LC_MESSAGES=C
    LC_MONETARY=ru_RU.CP1251
    LC_NUMERIC=ru_RU.CP1251
    LC_TIME=C
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/at/bin:/bin:/usr/bin:/usr/X11R6/bin:/usr/local/bin:/usr/games:/sbin:/usr/sbin:/usr/local/sbin:/usr/lib/qt3/bin
    PERL_BADLANG (unset)
    SHELL=/bin/zsh

```
</details>

**Comment by p5pRT** [comment]
### From rra@stanford.edu

Alexey Tourbin \(via RT\) \<perlbug\-followup@&#8203;perl\.org> writes&#8203;:

> When I run "perldoc LWP"\, I get wrong manpage title&#8203;: "vendor\_perl&#8203;::LWP"\.
> \(The same way I get "vendor\_perl&#8203;::XML&#8203;::Parser" for XML&#8203;::Parser etc\.\)
> This is because these modules are installed in vendor directories
> \(widely used in modern Linux distributions for vendor\-supplied perl
> packages\)\.  Vendor directories were introduced with perl\-5\.8\.0\, and some
> pieces of software are not ready for this change\.

> Here is a patch for Pod/Man\.pm that fixes the problem for me&#8203;:

> \-\-\- perl\-5\.8\.6/lib/Pod/Man\.pm\-    2003\-03\-31 11&#8203;:16&#8203;:36 \+0000
> \+\+\+ perl\-5\.8\.6/lib/Pod/Man\.pm    2005\-03\-19 18&#8203;:58&#8203;:12 \+0000
> @&#8203;@&#8203; \-411\,7 \+411\,7 @&#8203;@&#8203; sub begin\_pod \{
>              \}
>              if \($cut > 0\) \{
>                  splice \(@&#8203;dirs\, 0\, $cut\);
> \-                shift @&#8203;dirs if \($dirs\[0\] =~ /^site\(\_perl\)?$/\);
> \+                shift @&#8203;dirs if \($dirs\[0\] =~ /^\(site|vendor\)\(\_perl\)?$/\);
>                  shift @&#8203;dirs if \($dirs\[0\] =~ /^\[\\d\.\]\+$/\);
>                  shift @&#8203;dirs if \($dirs\[0\] =~ /^\(\.\*\-$^O|$^O\-\.\*|$^O\)$/\);
>              \}
> End of patch

Thanks\.  This fix will be included in the next release of Pod&#8203;::Man \(which
is still waiting on a new Pod&#8203;::Simple release\)\.

\-\- 
Russ Allbery \(rra@&#8203;stanford\.edu\)             \<http&#8203;://www\.eyrie\.org/~eagle/>

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @smpeters

On Sat Mar 19 11&#8203;:27&#8203;:54 2005\, at@&#8203;altlinux\.ru wrote&#8203;:
> This is a bug report for perl from at@&#8203;altlinux\.ru\,
> generated with the help of perlbug 1\.35 running under perl v5\.8\.6\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> Hello\,
> 
> When I run "perldoc LWP"\, I get wrong manpage title&#8203;:
> "vendor\_perl&#8203;::LWP"\.
> \(The same way I get "vendor\_perl&#8203;::XML&#8203;::Parser" for XML&#8203;::Parser etc\.\)
> This is because these modules are installed in vendor directories
> \(widely
> used in modern Linux distributions for vendor\-supplied perl packages\)\.
> Vendor directories were introduced with perl\-5\.8\.0\, and some pieces
> of software are not ready for this change\.
> 
> Here is a patch for Pod/Man\.pm that fixes the problem for me&#8203;:
> 
> \-\-\- perl\-5\.8\.6/lib/Pod/Man\.pm\-    2003\-03\-31 11&#8203;:16&#8203;:36 \+0000
> \+\+\+ perl\-5\.8\.6/lib/Pod/Man\.pm    2005\-03\-19 18&#8203;:58&#8203;:12 \+0000
> @&#8203;@&#8203; \-411\,7 \+411\,7 @&#8203;@&#8203; sub begin\_pod \{
>              \}
>              if \($cut > 0\) \{
>                  splice \(@&#8203;dirs\, 0\, $cut\);
> \-                shift @&#8203;dirs if \($dirs\[0\] =~ /^site\(\_perl\)?$/\);
> \+                shift @&#8203;dirs if \($dirs\[0\] =~
> /^\(site|vendor\)\(\_perl\)?$/\);
>                  shift @&#8203;dirs if \($dirs\[0\] =~ /^\[\\d\.\]\+$/\);
>                  shift @&#8203;dirs if \($dirs\[0\] =~ /^\(\.\*\-$^O|$^O\-\.\*|$^O\)$/\);
>              \}
> End of patch
> 
> Now I get correct manpage titles\.
> I also noticed similar issue in Pod/Find\.pm&#8203;:
> 
> \-\-\- perl\-5\.8\.6/lib/Pod/Find\.pm\-    2004\-06\-01 12&#8203;:41&#8203;:24 \+0000
> \+\+\+ perl\-5\.8\.6/lib/Pod/Find\.pm    2005\-03\-19 19&#8203;:10&#8203;:56 \+0000
> @&#8203;@&#8203; \-168\,10 \+168\,10 @&#8203;@&#8203; sub pod\_find
> 
>          if \($^O eq 'MacOS'\) \{
>              $SIMPLIFY\_RX =
> \-              qq\!^\(?i&#8203;:\\&#8203;:?site\_perl\\&#8203;:|\\&#8203;:?pod\\&#8203;:\(?=\.\*?\\\\\.pod\\\\z\)\)\*\!;
> \+
> qq\!^\(?i&#8203;:\\&#8203;:?\(?&#8203;:site|vendor\)\_perl\\&#8203;:|\\&#8203;:?pod\\&#8203;:\(?=\.\*?\\\\\.pod\\\\z\)\)\*\!;
>          \} else \{
>              $SIMPLIFY\_RX =
> \-
> qq\!^\(?i&#8203;:site\(\_perl\)?/|\\Q$Config&#8203;::Config\{archname\}\\E/|\\\\d\+\\\\\.\\\\d\+\(\[\_\.\]?\\\\d\+\)?/|pod/\(?=\.\*?
\\\\\.pod\\\\z\)\)\*\!;
> \+
> qq\!^\(?i&#8203;:\(?&#8203;:site|vendor\)\(\_perl\)?/|\\Q$Config&#8203;::Config\{archname\}\\E/|\\\\d\+\\\\\.\\\\d\+\(\[\_\.\]?\\\\d\+\)?
/|pod/\(?=\.\*?\\\\\.pod\\\\z\)\)\*\!;
>          \}
>      \}
> End of patch
> 
> This two files are known to be part of "podlators" and "PodParser"
> CPAN
> distributions\.  Should I file bugs at rt\.cpan\.org too?
> 
> 


This fix was included with podlators\-2\.00\.  Thanks\!

Steve Peters

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
